RFS/src/main/java/com/rfs/common/RestClient.java

```diff
@@ -28,108 +25,61 @@ public Response(int responseCode, String responseBody, String responseMessage) {
-        int responseCode = conn.getResponseCode();
+    public Response get(String path, boolean quietLogging) {
+        return getAsync(path, quietLogging).block();
+    }
 
-        String responseBody;
-        if (responseCode >= HttpURLConnection.HTTP_BAD_REQUEST) {
-            // Read error stream if present
-            try (BufferedReader br = new BufferedReader(new InputStreamReader(conn.getErrorStream(), StandardCharsets.UTF_8))) {
-                responseBody = br.lines().collect(Collectors.joining(System.lineSeparator()));
-            }
-        } else {
-            // Read input stream for successful requests
-            try (BufferedReader br = new BufferedReader(new InputStreamReader(conn.getInputStream(), StandardCharsets.UTF_8))) {
-                responseBody = br.lines().collect(Collectors.joining(System.lineSeparator()));
-            }
-        }
+    public Mono<Response> postAsync(String path, String body) {
+        return client.post()
+            .uri("/" + path)
+            .send(ByteBufMono.fromString(Mono.just(body)))
+            .responseSingle((response, bytes) -> bytes.asString()
+                .map(b -> new Response(response.status().code(), b, response.status().reasonPhrase())));
+    }
 
-        if (quietLogging || (responseCode == HttpURLConnection.HTTP_CREATED || responseCode == HttpURLConnection.HTTP_OK)) {
-            logger.debug("Response Code: " + responseCode + ", Response Message: " + conn.getResponseMessage() + ", Response Body: " + responseBody);
-        } else {
-            logger.error("Response Code: " + responseCode + ", Response Message: " + conn.getResponseMessage() + ", Response Body: " + responseBody);
-        }
+    public Mono<Response> putAsync(String path, String body, boolean quietLogging) {
+        return client.put()
+            .uri("/" + path)
+            .send(ByteBufMono.fromString(Mono.just(body)))
+            .responseSingle((response, bytes) -> bytes.asString()
+                .map(b -> new Response(response.status().code(), b, response.status().reasonPhrase()))
+                .doOnNext(resp -> logResponse(resp, quietLogging)));
+    }
 
-        conn.disconnect();
-        
-        return new Response(responseCode, responseBody, conn.getResponseMessage());
+    public Response put(String path, String body, boolean quietLogging) {
+        return putAsync(path, body, quietLogging).block();
     }
 
-    private void setAuthHeader(HttpURLConnection conn) {
-        if (connectionDetails.authType == ConnectionDetails.AuthType.BASIC) {
-            String auth = connectionDetails.username + ":" + connectionDetails.password;
-            byte[] encodedAuth = Base64.getEncoder().encode(auth.getBytes(StandardCharsets.UTF_8));
-            String authHeaderValue = "Basic " + new String(encodedAuth);
-            conn.setRequestProperty("Authorization", authHeaderValue);
+    private void logResponse(Response response, boolean quietLogging) {
```

Comment:
Should this be protected so that others can leverage the class with different policies on logging?

Reply:
I ripped it out in my local changes, consider this function gone.